### ISSUE TITLE
Running pre-commit hook from Xcode

### ISSUE THREAD

**Issue by Milos** [nicklockwood/SwiftFormat#670]
When trying to commit from Xcode, pre-commit tries to run swiftformat from the repo but _fails_ to commit. 

**Xcode warning message after commit:** 

```
The working copy “Testing” failed to commit files.
Running SwiftFormat...\n
SwiftFormat completed in 0.13s.
1/1 files formatted.
```

In the **Xcode 11.5**, it seems as the changes were formatted and committed but in the terminal it looks like the changes where committed and then formatted resulting in *unstaged* changes. 

**pre-commit hook**

`git diff --diff-filter=d --staged --name-only | grep -e '\(.*\).swift$' | while read line; do  
  $PWD/swiftformat "${line}";
  git add "$line";
done`

**Comment by nicklockwood** [comment]
@Milos have you tried using [git-format-staged](https://github.com/hallettj/git-format-staged) as suggested in the README?

**Comment by Milos** [comment]
Yes. Using **git-format-staged** _fixed_ the issue. However, my `.swiftformat` config file doesn't seem to work with wildcards 
For example: 
```
--header "ignore" **/File.swift
--exclude **/OtherFile.swift
```
After formatting, File's header becomes `// ignore **/File.swift` and `exclude` is ignored. On the other hand, command-line rules and options are working properly. 🤔

**Comment by nicklockwood** [comment]
`--header ignore **/File.swift` is not a supported syntax in the .swiftformat config file. All options or rules in the file are applied to every file formatted.

The `--exclude **/OtherFile.swift` should work, but it applies to all formatting rules.

**Comment by Milos** [comment]
Hmm, ok. I've checked again and exclude option is not working in the config file. 

Same happens if I use the options (header ignore and exclude) in the pre-commit hook: ```git-format-staged --formatter "swiftformat stdin --stdinpath {}" "*.swift"```. 

So what are other options in this case?

**Comment by nicklockwood** [comment]
I suspect the `.swiftformat` file is being ignored completely when you use `git-format-staged`, since it's reading from stdin. Try adding an explicit `--config` argument, e.g.

```
git-format-staged --formatter "swiftformat stdin --stdinpath {} --config .switformat" "*.swift"
```

**Comment by Milos** [comment]
I've already did. This is my pre-commit hook:

```$PWD/git-format-staged --formatter "$PWD/swiftformat --config $PWD/Format/.swiftformat stdin --stdinpath {}" "*.swift"```

I know for sure that config file is picked up because  `--header` option works. On the other hand, `--exclude` doesn't work at all...

**Comment by nicklockwood** [comment]
@Milos ah sorry, you're right. I never considered the `--stdinpath` in the exclude logic. That's a bug.

**Comment by Milos** [comment]
@nicklockwood , oh ok. Thanks! By the way, Is there any way around this at the moment (to ignore headers and exclude some file), maybe without using the config file?

**Comment by Milos** [comment]
@nicklockwood  by the way, is there a workaround at the moment? 

Passing header ignore and/or exclude option in the pre-commit hook using `git-format-staged` results in an error: `Cannot combine stdin with other file inputs.`

Example: `$PWD/git-format-staged --formatter "$PWD/swiftformat --header ignore **/File.swift stdin --stdinpath {}" "*.swift"`

**Comment by nicklockwood** [comment]
@Milos I'm not aware of any workaround in swiftformat. It looks like [git-format-staged](https://github.com/hallettj/git-format-staged) has a syntax for file exclusion, so maybe you can use that?

**Comment by Milos** [comment]
> @Milos I'm not aware of any workaround in swiftformat. It looks like [git-format-staged](https://github.com/hallettj/git-format-staged) has a syntax for file exclusion, so maybe you can use that?

Ok, I'll take a look, but as I understand, for now it's not possible to ignore headers using swiftformat in pre-commit hook?

**Comment by nicklockwood** [comment]
@Milos I believe you could do it by applying git-format-staged multiple times with different rules, something like:

```
$PWD/git-format-staged --formatter "$PWD/swiftformat --config .swiftformat --header ignore --stdinpath {}" "File.swift"
$PWD/git-format-staged --formatter "$PWD/swiftformat --config .swiftformat --stdinpath {}" "*.swift" "!File.swift"
```

**Comment by Milos** [comment]
> @Milos I believe you could do it by applying git-format-staged multiple times with different rules, something like:
> 
> ```
> $PWD/git-format-staged --formatter "$PWD/swiftformat --config .swiftformat --header ignore --stdinpath {}" "File.swift"
> $PWD/git-format-staged --formatter "$PWD/swiftformat --config .swiftformat --stdinpath {}" "*.swift" "!File.swift"
> ```

Did the trick. Thanks @nicklockwood 🙌

**Comment by nicklockwood** [comment]
@Milos fixed in 0.45.5.
